src/Raven.Server/Documents/CollectionRunner.cs

```diff
@@ -116,6 +116,8 @@ public virtual Task<IOperationResult> ExecuteDelete(string collectionName, long
                                     break;
                                 }
 
+                                context.Transaction.ForgetAbout(document);
```

Comment:
Should we use the `PulsedTransaction` here?

Reply:
That is relevant for compression only, on

Reply:
Now I'm wondering. Shouldn't this be after we materialize the Id in ids.Enqueue? We are releasing the buffer before we materialize the Id? Maybe it is not a problem?

Reply:
Yes, need to reverse this, to be safe

Reply:
Pushed: https://github.com/ravendb/ravendb/commit/14dc3a84c8d7256374ff3383723e45d367052779